zebra-consensus/src/transaction.rs

```diff
@@ -533,6 +540,26 @@ where
 
         Ok(async_checks)
     }
+
+    /// Verifies a transaction's Orchard shielded data.
+    fn verify_orchard_shielded_data(
+        orchard_shielded_data: &Option<orchard::ShieldedData>,
+        shielded_sighash: &blake2b_simd::Hash,
+    ) -> Result<AsyncChecks, TransactionError> {
+        let mut async_checks = AsyncChecks::new();
+
+        if let Some(orchard_shielded_data) = orchard_shielded_data {
+            let bvk = orchard_shielded_data.binding_verification_key();
+
+            async_checks.push(
+                primitives::redpallas::VERIFIER
+                    .clone()
+                    .oneshot((bvk, orchard_shielded_data.binding_sig, &shielded_sighash).into()),
+            );
```

Comment:
Assumes the async verifier will not fail but we maybe need to do the same as sapling. This is run themnot async and ignore the errors.

Reply:
If there are any validation errors, we should fix them, then merge this PR.

That probably means that this PR is blocked on the sapling value balance fix in #1939.